vector/src/main/scala/geotrellis/vector/reproject/Reproject.scala

```diff
@@ -100,7 +100,9 @@ object Reproject {
       val length = sqrt(pow(x0 - x1, 2) + pow(y0 - y1, 2))
 
       val p2 = m -> (x2, y2)
-      if (deflect / length < relError) {
+      if (deflect.isNaN) {
+        throw new IllegalArgumentException(s"Check the input $extent CRS, it is probably not in the correct projection.")
```

Comment:
Exception should indicate a little bit of mechanical reason why it was thrown because its super surprising every time it happens. Something like s"Encountered NaN during refinement step: ($deflect / $length). Input $extent is likely not in source projection." It would be great if we could pull out the actual CRS from the [`Transform`](https://github.com/locationtech/geotrellis/blob/master/proj4/src/main/scala/geotrellis/proj4/Transform.scala) instance here but still good enough to get a hint.

Reply:
Yea, it is impossible to get input and output CRS from the `Transform` here, since it in this case it is a Function:

```scala
type Transform = (Double, Double) => (Double, Double)
```